### PR TITLE
Limit RTAMO attribution to www referrals

### PR DESCRIPTION
**Pull request by moz-astults**

### CODE REVIEW

**Review by oremj — APPROVED** [review]
Nice! This looks good to me. Can we have https://github.com/muffinresearch review as well?

File: handlers.go [hunk 1]
```diff
@@ -390,11 +419,22 @@ func (b *BouncerHandler) shouldAttribute(reqParams *BouncerParams) bool {
 		}
 	}
 
+	// Check if the request is coming from RTAMO, and if so, only attribute
+	// if there is a referer header from https://www.mozilla.org/
+	// https://github.com/mozilla-services/go-bouncer/issues/347
+	if fromRTAMO(reqParams.AttributionCode) {
+		if reqParams.Referer != "https://www.mozilla.org/" {
```
Comment (muffinresearch) [inline handlers.go:426]: What about cases where the referer header includes a path?

File: handlers_test.go [hunk 2]
```diff
@@ -110,6 +110,27 @@ func TestShouldAttribute(t *testing.T) {
...
+			},
+			false,
+		},
+		{
+			&BouncerParams{
+				OS:              "win",
+				Product:         "firefox-stub",
+				AttributionCode: "c291cmNlPWFkZG9ucy5tb3ppbGxhLm9yZyZtZWRpdW09cmVmZXJyYWwmY2FtcGFpZ249bm9uLWZ4LWJ1dHRvbiZjb250ZW50PXJ0YTplMkk1WkdJeE5tRTBMVFpsWkdNdE5EZGxZeTFoTVdZMExXSTROakk1TW1Wa01qRXhaSDAmZXhwZXJpbWVudD0obm90IHNldCkmdmFyaWF0aW9uPShub3Qgc2V0KSZ1YT1lZGdlJnZpc2l0X2lkPShub3Qgc2V0KQ..",
+				AttributionSig:  "att-sig",
+				Referer:         "https://www.mozilla.org/",
```
Comment (muffinresearch) [inline handlers_test.go:130]: Would be good to cover a referer header containing paths in the tests as well.

File: handlers.go [hunk 3]
```diff
@@ -390,11 +419,22 @@ func (b *BouncerHandler) shouldAttribute(reqParams *BouncerParams) bool {
 		}
 	}
 
+	// Check if the request is coming from RTAMO, and if so, only attribute
+	// if there is a referer header from https://www.mozilla.org/
+	// https://github.com/mozilla-services/go-bouncer/issues/347
+	if fromRTAMO(reqParams.AttributionCode) {
```
Comment (muffinresearch) [inline handlers.go:425]: Presumably in go, reqParams are implicitly urldecoded?

**Review by muffinresearch — CHANGES_REQUESTED** [review]
I think this needs to cover the case where the referer header contains a path. Otherwise this looks good.

File: handlers.go (hunk 1 above)
Comment (moz-astults) [reply handlers.go]: Added a regex to match anything starting with `https://www.mozilla.org/`

File: handlers.go (hunk 3 above)
Comment (moz-astults) [reply handlers.go:425]: @muffinresearch `reqParams` are created in this function which takes the URL encoded parameters and sets (decodes) the values accordingly: https://github.com/mozilla-services/go-bouncer/blob/347/params.go#L21-L31

Inside of the attribution code parameter, there is a base64 encoded value that once decoded can be further URL decoded which happens in the `fromRTAMO` function.

Here's an example showing the different steps to decode the URL and get to the section where we can tell it's from RTAMO: https://play.golang.org/p/5FQf29j0P-C

**Comment by moz-astults** [comment]
@muffinresearch @oremj  Thanks for the feedback! I think I've addressed the concerns mentioned above and added another test, but please let me know if you see anything else!

**Review by muffinresearch — APPROVED** [review]
lgtm

**Pull request merged**